client/apps/edit/components/content/section_controls/test/index.coffee

```diff
@@ -0,0 +1,142 @@
+      component.setInsideComponent()
+      component.state.insideComponent.should.eql true
+
+
+  describe '#getControlsWidth', ->
+
+    it 'returns 620 for column_width sections', ->
+      @component.props.section.set 'layout', 'column_width'
+      width = @component.getControlsWidth()
+      width.should.eql 620
+
+    it 'returns 900 if image_set or has overflow layout', ->
+      width = @component.getControlsWidth()
+      width.should.eql 900
+
+      @component.props.section.set 'type', 'image_set'
+      width = @component.getControlsWidth()
+      width.should.eql 900
+
+      @component.props.section.set 'type', 'video'
+      width = @component.getControlsWidth()
+      width.should.eql 900
+
+    it 'returns 1100 if hero section', ->
+      @props.isHero = true
+      component = ReactDOM.render React.createElement(@SectionControls, @props), (@$el = $ "<div></div>")[0], =>
+      width = component.getControlsWidth()
+      width.should.eql 1100
+
+
+  describe '#getHeaderSize', ->
+
+    it 'returns 95 when channel is editorial', ->
+      header = @component.getHeaderSize()
+      header.should.eql 95
+
+    it 'returns 55 when channel is not editorial', ->
+      @props.channel = { isEditorial: sinon.stub().returns(false) }
+      component = ReactDOM.render React.createElement(@SectionControls, @props), (@$el = $ "<div></div>")[0], =>
+      header = component.getHeaderSize()
+      header.should.eql 55
+
+
+  describe '#getPositionBottom', ->
+
+    it 'when inside component, calculates based on window scroll position', ->
+      @component.props.section.set 'type', 'video'
+      @component.setState insideComponent: true
+      bottom = @component.getPositionBottom()
+      bottom.should.eql 605
+
+    it 'when inside component, adds a 20px buffer for images', ->
+      @component.setState insideComponent: true
+      bottom = @component.getPositionBottom()
+      bottom.should.eql 585
+
+    it 'when outside component, returns 100%', ->
+      @component.props.section.set 'type', 'video'
+      @component.setState insideComponent: false
+      bottom = @component.getPositionBottom()
+      bottom.should.eql '100%'
+
+    it 'when outside component, adds a 20px buffer for images', ->
+      @component.setState insideComponent: false
+      bottom = @component.getPositionBottom()
+      bottom.should.eql 'calc(100% + 20px)'
+
+
+  describe '#getPositionLeft', ->
+
+    it 'calculates width based on window when insideComponent', ->
+      left = @component.getPositionLeft()
+      left.should.eql 305
+
+    it 'returns 0 if outside component', ->
+      @component.props.section.set 'type', 'video'
+      @component.setState insideComponent: false
+      left = @component.getPositionLeft()
+      left.should.eql 0
+
+    it 'adds a 20px buffer for images', ->
+      @component.setState insideComponent: false
+      left = @component.getPositionLeft()
+      left.should.eql -20
```

Comment:
👍  set of tests -- glad you were able to break out that `insideComponent` function and find a more testable approach.